version: "3.5"

volumes:
  gitea_web:
    driver: local
  gitea_db:
    driver: local

networks:
  gitea:
    external: false

services:
  app:
    image: gitea/gitea:1.16.8
    container_name: gitea-web
    restart: always
    ports:
      - ${PORT}:3000
      - 222:22
    environment:
      - USER_UID=1000
      - USER_GID=1000
      - GITEA__database__DB_TYPE=postgres
      - GITEA__database__HOST=db:5432
      - GITEA__database__NAME=${GITEA_DATABASE}
      - GITEA__database__USER=${GITEA_USER}
      - GITEA__database__PASSWD=${GITEA_DATABASE_PASSWORD}
    networks:
      - gitea
    volumes:
      - gitea_web:/data
      - /etc/timezone:/etc/timezone:ro
      - /etc/localtime:/etc/localtime:ro
    depends_on:
      - db

  db:
    image: postgres:14
    container_name: gitea-db
    restart: always
    ports:
      - ${GITEA_DATABASE_PORT}:5432
    environment:
      - POSTGRES_USER=${GITEA_USER}
      - POSTGRES_PASSWORD=${GITEA_DATABASE_PASSWORD}
      - POSTGRES_DB=${GITEA_DATABASE}
    networks:
      - gitea
    volumes:
      - gitea_db:/var/lib/postgresql/data
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U ${GITEA_USER}"]
      interval: 5s
      timeout: 5s
      retries: 5


